### PR TITLE
ONEXL-695-timeslots-v-3-proposal

### PR DESCRIPTION
**Pull request by joaotnlima**
## Summary
V2 timeslots following Glovo M1 standards.
This is just a proposal to be reviewed on partner side.

## Related JIRA issues
[ONEXL-695](https://glovoapp.atlassian.net/browse/ONEXL-695)

## Checklist before requesting a review
- [ ] I have tested my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have included tests for my changes.

[ONEXL-695]: https://glovoapp.atlassian.net/browse/ONEXL-695?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: retailer-oas/paths/timeslots/timeslots.yaml [hunk 1]
```diff
@@ -39,22 +32,64 @@ get:
       example: 1
       schema:
         type: number
+    - name: service
```
Comment (deivi98) [inline retailer-oas/paths/timeslots/timeslots.yaml:35]: question: what is this about?

File: retailer-oas/paths/timeslots/timeslots.yaml [hunk 2]
```diff
@@ -39,22 +32,64 @@ get:
...
+    - name: lon
+      in: query
+      schema:
+        type: number
+      format: ISO 6709:2022
+      example: 139.94637467
+      description: Longitude in decimal degrees
     - name: postalCode
       in: query
+      description: The postal code of the shipping address
```
Comment (deivi98) [inline retailer-oas/paths/timeslots/timeslots.yaml:57]: question: only `postalCode` and `address` will be required?

File: retailer-oas/paths/timeslots/timeslots.yaml (hunk 1 above)
Comment (joaotnlima) [reply retailer-oas/paths/timeslots/timeslots.yaml:35]: A prep for what will come in the future. The possibility of filtering timeslots/PLs that offer different services if needed.

File: retailer-oas/paths/timeslots/timeslots.yaml (hunk 2 above)
Comment (joaotnlima) [reply retailer-oas/paths/timeslots/timeslots.yaml:57]: Postal code, addressline1, addressline2,  countryCode.
This is a proposal... the intention is for them to only use lon and lat... if that fails we need all these fields to fetch google maps API for the same info (lon, lat).

**Review by JordiArmengol — APPROVED** [review]

**Review by rob-molero — APPROVED** [review]

File: retailer-oas/schemas/Timeslots.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                capacity:
+                  $ref: "./Capacity.yaml"
+                availability:
+                  type: boolean
+                  description: Indicates whether this Timeslot is available to be reserved or not.
+                  example: true
+  pagination:
+    type: object
+    properties:
+      limit:
```
Comment (mary-hg) [inline retailer-oas/schemas/Timeslots.yaml:34]: question: limit and page aren't query params?

Comment (joaotnlima) [reply retailer-oas/schemas/Timeslots.yaml:34]: you're right. I will change the quantity param for the limit.
adding the page as well.

File: retailer-oas/paths/timeslots/timeslots.yaml [hunk 4]
```diff
@@ -11,13 +11,6 @@ get:
   tags:
     - timeslots
   parameters:
-    - name: pickingLocationId
```
Comment (mary-hg) [inline retailer-oas/paths/timeslots/timeslots.yaml:14]: todo: please is no pl id is involved update the description

File: retailer-oas/paths/timeslots/timeslots.yaml [hunk 5]
```diff
@@ -39,22 +32,64 @@ get:
...
       description: OK
       content:
         application/json:
           schema:
-            type: array
-            items:
-              $ref: '../../schemas/TimeslotAvailability.yaml'
+            type: object
+            $ref: '../../schemas/Timeslots.yaml'
     '400':
```
Comment (mary-hg) [inline retailer-oas/paths/timeslots/timeslots.yaml:100]: todo: update the response codes

File: retailer-oas/schemas/TimeslotAvailability.yaml [hunk 6]
```diff
@@ -1,10 +1,29 @@
...
+  data:
+    type: object
     properties:
-      capacity:
-        $ref: "./Capacity.yaml"
-      availability:
-        type: boolean
-        description: Indicates whether this Timeslot is available to be reserved or not.
-        example: true
+      pickingLocationRef:
```
Comment (mary-hg) [inline retailer-oas/schemas/TimeslotAvailability.yaml:6]: question: Can we have more than one pl per address? If we are returning the closets one please specify in description

Comment (joaotnlima) [reply retailer-oas/schemas/TimeslotAvailability.yaml:6]: did not understand? can you make a recommendation so I get what you want to change?

File: retailer-oas/paths/timeslots/timeslots.yaml (hunk 5 above)
Comment (joaotnlima) [reply retailer-oas/paths/timeslots/timeslots.yaml]: done. TY for the warning 👍

File: retailer-oas/paths/timeslots/timeslots.yaml (hunk 4 above)
Comment (joaotnlima) [reply retailer-oas/paths/timeslots/timeslots.yaml:14]: pls verify

File: retailer-oas/paths/timeslots/timeslots.yaml (hunk 5 above)
Comment (joaotnlima) [reply retailer-oas/paths/timeslots/timeslots.yaml]: pls verify

File: retailer-oas/schemas/TimeslotAvailability.yaml (hunk 6 above)
Comment (mary-hg) [reply retailer-oas/schemas/TimeslotAvailability.yaml:6]: I'm asking if it is possible to have more than one picking location for the same address, if it isn't possible no change is needed, but if it's possible and we are returning only one just add to description what should be returned. If we can return more than one pl for this get probably we need something like this in response `{
   "data":{
      "pickingLocations":[
         {
            "pickingLocationRef":"7996",
            "serviceType":{
               "pickDelivery":true,
               "pickCollect":true
            },
            "timeslots":[
               {
                  "id":"3fa85f64-5717-4562-b3fc-2c963f66afa6",
                  "fromDate":"2023-01-11T14:00:00+00:00",
                  "toDate":"2023-01-11T15:00:00+00:00",
                  "capacity":{
                     "occupancyRate":70
                  },
                  "availability":true
               }
            ]
         }
      ],
      "pagination":{
         "limit":100,
         "page":1,
         "total":100
      }
   }
}`

Comment (joaotnlima) [reply retailer-oas/schemas/TimeslotAvailability.yaml:6]: only one for the time being.

Comment (mary-hg) [reply retailer-oas/schemas/TimeslotAvailability.yaml:6]: or 
```
{
   "data":{
      "timeslots":[
         {
            "id":"3fa85f64-5717-4562-b3fc-2c963f66afa6",
            "fromDate":"2023-01-11T14:00:00+00:00",
            "toDate":"2023-01-11T15:00:00+00:00",
            "capacity":{
               "occupancyRate":70
            },
            "availability":true,
            "pickingLocations":[
               {
                  "pickingLocationRef":"7996",
                  "serviceType":{
                     "pickDelivery":true,
                     "pickCollect":true
                  }
               }
            ]
         }
      ],
      "pagination":{
         "limit":100,
         "page":1,
         "total":100
      }
   }
}
```

**Review by mary-hg — APPROVED** [review]

**Pull request merged**